javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,21 @@
+
+var i = -1;
+
+document.addEventListener('keydown', function (event) {
+
+  var $span = document.querySelectorAll('span');
+
+  i++;
```

Comment:
Instead of doing this in every situation, you should only move the current index forward if the current letter is correct. That will also remove the need to decrement your value if the letter typed is incorrect, and it will allow you to start `i` at `0`, which is the correct initial value for indexes in an array